substrate/client/rpc-spec-v2/src/common/events.rs

```diff
@@ -105,23 +105,41 @@ pub struct StorageResultErr {
 
 /// The result of a storage call.
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
-#[serde(untagged)]
-pub enum ArchiveStorageResult {
+#[serde(rename_all = "camelCase")]
+#[serde(tag = "event")]
+pub enum ArchiveStorageEvent {
 	/// Query generated a result.
-	Ok(ArchiveStorageMethodOk),
+	StorageResult(StorageResult),
 	/// Query encountered an error.
-	Err(ArchiveStorageMethodErr),
+	StorageErr(ArchiveStorageMethodErr),
+	/// Operation storage is done.
+	StorageDone,
```

Comment:
Shouldn't this match the spec (https://paritytech.github.io/json-rpc-interface-spec/api/archive_unstable_storage.html#storagedone) ie in the spec we have 3 events, "storage" (getting value back), "storageDone" (finished) and "storageError" for errors?